README.md

```diff
@@ -19,6 +19,137 @@ it, simply add the following line to your Podfile:
 
     pod 'MetovaBase'
 
+## Overview
+
+MetovaBase offers useful UIKit subclasses that implement commonly used utilities.
+
+## BaseViewController
+
+The `BaseViewController` class extends from `UIViewController` and adds useful, common utility that we have found ourselves adding in most, if not all, of our view controller classes.
+It does some additional set up in `viewDidLoad` to provide your view controller with keyboard notifications.  If overriding, be sure to call `super.viewDidLoad` in your subclass.
+
+### Example usage:
+
+```swift
+import MetovaBase
+
+class MyViewController: BaseViewController {
+
+    override func viewDidLoad() {
+    
+    	// this call registers your VC for keyboard notifications
+    	super.viewDidLoad() 
+    	
+    	// my custom setup here
+    }
+}
+```
+
+Additionally, `BaseViewController` unregisters your view controller from receiving keyboard notifications in its `deinit()` method.
+
+## KeyboardNotificationDelegate
+
+This protocol contains a collection of methods which `BaseViewController` subclasses (or extensions) can implement in order to hear about keyboard notifications. As already stated, you do not need to register or unregister for these notifications in your subclass or extension--you simply need to implement these methods.
+
+`BaseViewController` uses the `respondsToSelector:` method on itself to determine whether or not it should register for these notifications. 
+ As such, if implementing these in Swift, you must mark them as either `@objc` or `dynamic`.  If not, they are not implemented as selectors and `respondsToSelector:` returns `false`.
```

Comment:
Yea, I haven't done that before either. It was something I was hoping to discuss. I don't think it's needed